src/app/client_sdk/poseidon_hash.ml

```diff
@@ -0,0 +1,138 @@
+open Core_kernel
+open Js_of_ocaml
+
+(* *************** *
+ *    our field    *
+ * *************** *)
+
+module Field = struct
+  include Snark_params_nonconsensus.Field
+
+  (* Converts a byterray into a [Field.t], raises an exception if the number obtained is larger than the order *)
+  let of_bytes bytearray =
+    let aux i acc c =
+      let big = Nat.of_int @@ int_of_char c in
+      let offset = Nat.shift_left big (Int.( * ) i 8) in
+      Nat.(acc + offset)
+    in
+    let zero = Nat.of_int 0 in
+    let big = Array.foldi bytearray ~init:zero ~f:aux in
+    let one = Nat.of_int 1 in
+    if Nat.(order - one < big) then
+      failwith "the given field is larger than the order" ;
+    of_bigint big
+
+  (* Converts a field element into an hexadecimal string (encoding the field element in little-endian) *)
+  let to_hex field =
+    (* taken from src/lib/pickles *)
+    let bits_to_bytes bits =
+      let byte_of_bits bs =
+        List.foldi bs ~init:0 ~f:(fun i acc b ->
+            if b then acc lor (1 lsl i) else acc )
+        |> Char.of_int_exn
+      in
+      List.map
+        (List.groupi bits ~break:(fun i _ _ -> i mod 8 = 0))
+        ~f:byte_of_bits
+      |> String.of_char_list
+    in
+    let bytearray = to_bits field |> bits_to_bytes in
+    Hex.encode bytearray
+end
+
+(* ********************** *
+ * our permutation Config *
+ * ********************** *)
+
+module Config = struct
+  module Field = Field
+
+  let rounds_full = 54
```

Comment:
Should this PR be against `develop`? I think we want to avoid including this in the client SDK until we've transitioned to this configuration for the main hash function too, which will be a breaking change that we'll want to land there.

Reply:
maybe a good time to ask: is there some documentation on how we manage different branches and create releases?

Reply:
For branches, the current model is:
* anything that is protocol-compatible with the current mainnet goes in `compatible`
* anything that needs an associated hard-fork goes in `develop`

There's some more nuance about how and when to land things, and how the released work; I think there's a notion document somewhere with the details.

Reply:
and is develop rebased on compatible periodically? Also what about master since it is the default branch?

Reply:
Compatible should be merged into develop, ideally straight after merging each PR into compatible.

Reply:
What's the conclusion on this :D ? I'm fine with either way